src/framework/depth-layer.js

```diff
@@ -0,0 +1,261 @@
+import {
+    ADDRESS_CLAMP_TO_EDGE,
+    CLEARFLAG_COLOR, CLEARFLAG_DEPTH,
+    FILTER_NEAREST,
+    PIXELFORMAT_DEPTHSTENCIL, PIXELFORMAT_R8_G8_B8_A8
+} from '../graphics/constants.js';
+
+import { RenderTarget } from '../graphics/render-target.js';
+import { Texture } from '../graphics/texture.js';
+
+import {
+    LAYERID_DEPTH, LAYERID_WORLD,
+    SHADER_DEPTH
+} from '../scene/constants.js';
+
+import { Layer } from '../scene/layer.js';
+
+// Internal class abstracting the access to the depth texture of the scene.
+// For webgl 2 devices, actual depth buffer is copying to a texture at specified time during camera rendering
+// for webgl 1 devices, the scene's depth is rendered to a separate RGBA texture
+class DepthLayer {
```

Comment:
wondering about this naming. it's not a layer?

Reply:
yeah it's not a Layer .. it's equivalent to a Picker or post PostEffectQueue, but internally all it does it creates a special Layer with name "Depth"

Reply:
`DepthManager`?

Reply:
I considered that very name .. but it implies a depth manager does a lot more than just copy out a depth buffer.
Maybe CameraDepthBuffer or CameraDepth?

Reply:
or SceneDepth?

Reply:
I'd be happy with SceneDepth.

Reply:
done, good name.